src/codegen/doc/mod.rs

```diff
@@ -254,12 +254,7 @@ fn create_object_doc(w: &mut dyn Write, env: &Env, info: &analysis::object::Info
                 .into_iter()
                 .chain(env.class_hierarchy.subtypes(info.type_id))
                 .filter(|&tid| !env.type_status(&tid.full_name(&env.library)).ignored())
-                .map(|tid| {
-                    format!(
-                        "[`{name}`](struct.{name}.html)",
-                        name = env.library.type_(tid).get_name()
-                    )
-                })
+                .map(|tid| format!("[`{n}`](crate::{n})", n = env.library.type_(tid).get_name()))
```

Comment:
I would suggest to use `"[`{n}`](struct@crate::{n})"` everywhere.

If `crate::x` does not exist, rustdoc dumps `href="crate::x"` into the html code which is in turn motivates browser to use xdg-open on this. With `struct@` it just links to an empty page.

Reply:
@sophie-h that's interesting, I don't get any links in GStreamer-rs for missing references at all: do you have an example? Unfortunately all our uses of this code sits within `glib::wrapper` and does not get warnings for broken references, making it hard to find one here.

That said, besides improved linking it seems like a good idea to explicitly specify the type with `struct@`, thanks for the suggestion! Done the same for `trait@` and updated the PR :grin:

Reply:
I only managed to get broken links by not having the newest gtk version enabled for `cargo doc` and then having links to objects from more recent versions from the LGPL docs I guess?

The online published version of gtk-rs shouldn't have any gir generated broken links because I wasn't getting any warning with all features enabled. So this is really just a precaution.

Reply:
> I wasn't getting any warning with all features enabled

It doesn't seem like there are much if any mistakes in GStreamer (so far only `gst::ObjectExt` should have been `gst::GstObjectExt`, that's not really the pattern we should follow). I bet `gtk` also only has these links generated inside the `glib::wrapper!` macro hence no warnings show up.

Reply:
The LGPL docs are copied from another repo into the actual source code in gtk-rs before generating the docs.

Reply:
Similar in GStreamer, the docs are committed in the repo and built into a separate crate. Either way we'll only get warnings once the documentation is actually embedded in the code and built by `rustdoc`, and we won't see them when they're inside macro expansions :/

Reply:
Just to clarify: I regenerated the LGPL docs with your branch and then generated the gtk-rs docs based on that to test this PR.

Reply:
Great, good to know - thanks!